lib/cli/src/Cardano/CLI.hs

```diff
@@ -296,6 +331,202 @@ runCli = join . customExecParser preferences
   where
     preferences = prefs showHelpOnEmpty
 
+{-------------------------------------------------------------------------------
+                            Commands - 'HD Derivation'
+-------------------------------------------------------------------------------}
+
+cmdKey :: Mod CommandFields (IO ())
+cmdKey = command "key" $ info (helper <*> cmds) $ mempty
+    <> progDesc "Derive keys from mnemonics."
+  where
+    cmds = subparser $ mempty
+        <> cmdRootKey
+
+-- | Parse a key type
+--
+-- Note that we in the future might replace the type @ByronWalletStyle@ with
+-- another type, to include Shelley keys.
```

Comment:
Actually yes. I don't understand why all the functions and types are limited to "Byron". Although, "Shelley" key are simply generated following the Icarus' generation style, so they are to-so-speak supported but functions should not imply that this is only a byron thing, because it clearly isn't.

Reply:
>  I don't understand why all the functions and types are limited to "Byron". 

I believe you specifically said not to support `Shelley` keys here.

With that in mind, I didn't want to unnecessarily create a new type, since `ByronWalletStyle` already existed.

The parser is now called `walletStyleOption` (not mentioning byron.) Maybe it is only a matter of renaming `byronCliKeyScheme` to say `newCliKeyScheme`?

Reply:
Meh. Yes but Shelley /= Jormungandr, although Jörmungandr uses Icarus' key style so it's already covered...  to some extend, if one knows that one can use `--icarus` to generate "Jörmungandr keys". 

>  it is only a matter of renaming byronCliKeyScheme to say newCliKeyScheme?

Yes.

Reply:
Done. Now `newCliKeyScheme`.